File: EaterSpring/src/main/kotlin/xyz/purema/eater/interfaces/IProductOrderService.kt
package xyz.purema.eater.interfaces

import xyz.purema.eater.dtos.ProductOrderDto

interface IProductOrderService {
    fun getById(id: String): ProductOrderDto
    fun getAllOrders(): List<ProductOrderDto>
    fun getByCustomerId(customerId: String): List<ProductOrderDto>
    fun getByProductId(productId: String): List<ProductOrderDto>
}

File: EaterNet/Eater/Interfaces/IProductOrderService.cs
using Eater.Dtos;

namespace Eater.Interfaces
{
    public interface IProductOrderService
    {
        public ValueTask<ProductOrderDto> GetById(string id);
        public Task<List<ProductOrderDto>> GetAll();
        public Task<List<ProductOrderDto>> GetAllByCustomerId(string customerId);
        public Task<List<ProductOrderDto>> GetAllByProductId(string productId);
    }
}
